p34/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+use std::ops;
+
+#[derive(Debug)]
+struct BigUint4096 {
+    values: [u64; 64],
+}
+
+impl PartialEq for BigUint4096 {
+    fn eq(&self, other: &Self) -> bool {
+        for i in 0..64 {
+            if self.values[i] != other.values[i] {
+                return false;
+            }
+        }
+        true
+    }
+}
+
+impl Eq for BigUint4096 {}
+
+impl ops::Add<BigUint4096> for BigUint4096 {
+    type Output = BigUint4096;
+
+    fn add(self, _rhs: BigUint4096) -> BigUint4096 {
+        let mut values = self.values;
+        // for i in 0..64 {
+        let mut carry = false;
+        for (i, item) in values.iter_mut().enumerate() {
+            let mut carry_from_first = false;
+            let carry_from_second;
+            if carry {
+                (*item, carry_from_first) = item.overflowing_add(1);
+            }
+            (*item, carry_from_second) = item.overflowing_add(_rhs.values[i]);
+            carry = carry_from_first || carry_from_second;
+        }
+        if carry {
+            panic!("attempt to add with overflow")
+        }
+        BigUint4096 { values }
+    }
+}
+
+impl From<u64> for BigUint4096 {
+    fn from(value: u64) -> Self {
+        let mut values = [0; 64];
+        values[0] = value;
+        Self { values }
+    }
+}
+
+impl From<Vec<u64>> for BigUint4096 {
```

Comment:
It's better to use `TryFrom` here and return an error if the vector is too big.

Reply:
solved